pkg/localstorev2/storer.go

```diff
@@ -449,6 +461,34 @@ func (db *DB) Close() error {
 	return err
 }
 
+func (db *DB) SetRetrievalService(r retrieval.Interface) {
+	db.retrieval = r
+}
+
+func (db *DB) SetSyncer(s SyncReporter) {
+	db.lock.Lock(lockKeySetSyncer)
```

Comment:
why do we need a look for this? this should be a one time operation.

Reply:
Maybe use `sync.Once`.